### ISSUE TITLE
Array handling is not working as it should in `$or` and `$in`

### ISSUE THREAD

**Issue by vladimir-lu** [DataGrip/mongo-jdbc-driver#7]
In IntelliJ with driver version 1.7.3, the following query executes on mongo shell, but fails in IntelliJ

```
db.pages.find({
    $or: [
        {'page.tag': {$in: ['a', 'b', 'c']}}
    ]
});
```
With error code 2 and error message `$in needs an array`.

Let me know if you need any more info :)

**Comment by kornilova203** [comment]
Hello @vladimir-lu,
Thank you
This is a known issue https://youtrack.jetbrains.com/issue/DBE-10519
I work on embedding new shell to the driver [mongodb-js/mongosh](https://github.com/mongodb-js/mongosh)
It will fix this and many other problems
Actually I plan to release it this week! I'll notify you when it happens

**Comment by kornilova203** [comment]
Fixed by #8
